tests/unit/integrations_kafka_consumer.cpp

```diff
@@ -178,13 +178,7 @@ TEST_F(ConsumerTest, BatchInterval) {
 TEST_F(ConsumerTest, StartStop) {
   Consumer consumer{CreateDefaultConsumerInfo(), kDummyConsumerFunction};
 
-  auto start = [&consumer](const bool use_conditional) {
-    if (use_conditional) {
-      consumer.StartIfStopped();
-    } else {
-      consumer.Start();
-    }
-  };
+  auto start = [&consumer]() { consumer.Start(); };
 
   auto stop = [&consumer](const bool use_conditional) {
```

Comment:
I think this can be eliminated also. Why have you kept it? Just forgot it, or I missed something?

Reply:
Short version: it checks that both `Consumer::Stop` and `Consumer::StopIfRunning`work as expected.
Longer version: my understanding is that there are two flavors to stop a stream:
- `Consumer::Stop`: 
  - Raises an exception if the stream is not running. 
  - Stops the stream if it is running.
  - Available to the outside world.
-`Consumer::StopIfRunning`: this stops the stream if it is running. 
  - Stops the stream if it is running.
  - Available to the outside world.
  - Only used in the destructor of `Consumer` and some unit tests.
`StopIfRunning` could be removed if we move its logic directly in the desctructor, then there is no need to check the `StopIfRunning` flavor and this test can be simplified.

Reply:
I missed its usage in the destructor. Then it is fine.